### PR TITLE
fix: suppress false-positive wrapper warning for game executables

### LINKED ISSUE #330: False-positive wrapper warning fires on normal game exit
## Summary

When a game process exits shortly after launch (user closed it, or it finished), SimLauncher incorrectly creates a persistent `processNameMismatchWarning` — as if a wrapper leaked a child process. The warning should only fire for **utility apps** where an early exit signals a wrapper pattern.

## Root cause

The `exit` handler in `spawn.ts` checks `exitedDuringPostLaunchWindow && !wasClosedBySimLauncher` but does not distinguish between a **game** and a **utility app**. Both trigger the warning on early exit.

## Expected behavior

- **Utility app** exits shortly after launch → warning fires (wrapper pattern suspected)
- **Game** exits shortly after launch → no warning (legitimate lifecycle event)

## Fix

Add an `isGame` check in the `exit` handler. Only emit the mismatch warning when the exiting process is a utility/custom app, not the main game executable.

Identified in internal audit `v0.9.6-testing-gemini` (PROC-001).

### LINKED ISSUE #351: State leak: process mismatch warnings never expire (missing expiresAt)
## Problem

Process mismatch warnings created in `spawn.ts` never expire because `expiresAt` is not set on the warning entry. The pruning function in `state.ts:33` correctly checks `entry.expiresAt !== undefined`, but the producer in `spawn.ts:347` never provides the field.

## Evidence

**Producer** (`spawn.ts:347-352`):
```typescript
processNameMismatchWarnings.set(appPath.toLowerCase(), {
  path: appPath,
  name: path.basename(appPath),
  gameKey,
  warning
  // ← Missing: expiresAt
})
```

**Consumer** (`state.ts:33-38`):
```typescript
export function pruneExpiredProcessNameMismatchWarnings(now = Date.now()) {
  processNameMismatchWarnings.forEach((entry, key) => {
    if (entry.expiresAt !== undefined && entry.expiresAt <= now) {
      processNameMismatchWarnings.delete(key)
    }
  })
}
```

**Type** (`types.ts:49`): `expiresAt?: number` — field exists but is optional, so TypeScript doesn't enforce it.

## Impact

- Warnings persist indefinitely until app restart
- Pollutes the "Close Apps" logic — stale warnings cause unnecessary UI state
- Confirmed independently by both Gemini and Opus audits (PROC-002, 1.12, 2.6)

## Fix

Add `expiresAt: Date.now() + TTL_MS` when creating the warning entry. A reasonable TTL would be the `POST_LAUNCH_BLOCK_MS` window or a dedicated constant (e.g. 60s).

## Source

- [[v0.9.6-testing-gemini]] PROC-002
- [[v0.9.6-testing-synthesis]] Critical Defects P1

### PR DESCRIPTION
**Pull request by Shieldxx**
## Summary

Two related fixes for the process mismatch warning system:

1. **Suppress false-positive warnings for game executables** — The "process name mismatch" warning was incorrectly firing when game-specific launchers (e.g. BeamNG.drive) exited shortly after spawning the main game process.

2. **Add TTL to prevent infinite warning lifetime** — Mismatch warnings were created without an `expiresAt` timestamp, making them immune to the pruning logic in `state.ts`. Warnings now expire after 60 seconds.

### Fix 1: Game-exe false positive (#330)
- Check `isGame` flag on `RunningProcessEntry` before triggering the wrapper/child warning
- Only fire the warning for utility apps, not game launchers
- Add regression test for game-exe early exit scenario
- Update existing adoption test to correctly use a non-game utility app

### Fix 2: Warning TTL (#351)
- Add `PROCESS_NAME_MISMATCH_WARNING_TTL_MS` constant (60s)
- Set `expiresAt` when creating warning entries in `spawn.ts`
- Add regression test verifying exact TTL boundary behavior
- Adjust existing wrapper test timing to respect the TTL window

Closes #330
Closes #351

Closes #330
Closes #351

### CODE REVIEW

File: src/main/processes/spawn.ts [hunk 1]
```diff
@@ -335,11 +335,13 @@ function spawnDetachedApp(
...
 
       child.once('exit', () => {
+        const processEntry = runningProcesses.get(appPath)
+        const wasGame = processEntry?.isGame ?? false
         runningProcesses.delete(appPath)
         const exitedDuringPostLaunchWindow = Date.now() - launchStartedAt <= POST_LAUNCH_BLOCK_MS
         const wasClosedBySimLauncher = consumeProcessNameMismatchWarningSuppression(appPath)
 
-        if (exitedDuringPostLaunchWindow && !wasClosedBySimLauncher) {
+        if (exitedDuringPostLaunchWindow && !wasClosedBySimLauncher && !wasGame) {
```
[the remark this answers is not in the document]
Comment (Shieldxx) [reply src/main/processes/spawn.ts]: Addressed in 3f46fba — game wrappers now create a **silent** mismatch entry (preserving `launchedGameKeys` for tracked adoption) but suppress the user-facing IPC notification. The entry auto-expires via the 60s TTL added in b16fb4a.

**Comment by Shieldxx** [comment]
@codex review

**Comment by Shieldxx** [comment]
Addressed in 5c07ae9 — game wrapper entries now omit `expiresAt` entirely, so they persist as adoption markers until the tracked child process exits naturally. Only utility entries get the 60s TTL. Proper parent-to-child process tracking is deferred to #325.
